routes/ci.js

```diff
@@ -78,47 +78,55 @@ router.post('/register', function (req, res, next) {
 })
 
 router.post('/webhook', function(req, res, next) {
-  var event = req.get('X-GitHub-Event')
-  switch (event) {
-    case "push":
-      repositoryModel.findOneRepository(
-        {
-          githubId: req.body.repository.owner.id,
-          name: req.body.repository.name
-        }
-      ).
-      then(function(result) {
-        var data = {
-          email: result.email,
-          gitUrl: req.body.repository.clone_url,
-          docTheme: "",
-        }
-        generator.executeScript({}, data, function(err, generatedData) {
-          if (err) {
-            console.log(err);
-          } else {
-            deploy.deployPages({}, {
-              email: result.email,
-              gitURL: req.body.repository.clone_url,
-              username: result.username,
-              uniqueId: generatedData.uniqueId,
-              encryptedToken: result.accessToken
-            })
+  var event = req.get('X-GitHub-Event');
+  var branch = req.body.ref.split("/")[2];
+  if (branch === "master") {
```

Comment:
No you are not! I want to maintain the flow of development. I didn't request a change that involves branch specific build. This change will cover the ground unless https://github.com/fossasia/yaydoc/issues/251 is resolved.